processor/groupbyattrsprocessor/processor_test.go

```diff
@@ -318,3 +391,18 @@ func someLogs(attrs pdata.AttributeMap, count int) pdata.Logs {
 
 	return logs
 }
+
+func someMetrics(attrs pdata.AttributeMap, count int) pdata.Metrics {
+	metrics := pdata.NewMetrics()
+	ilm := metrics.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
+
+	for i := 0; i < count; i++ {
+		metric := ilm.Metrics().AppendEmpty()
+		metric.SetName(fmt.Sprint("foo-", i))
+		metric.SetDataType(pdata.MetricDataTypeGauge)
+		dataPoint := metric.Gauge().DataPoints().AppendEmpty()
```

Comment:
Not sure but looks like we're missing tests for non-gauge types? Or maybe I just didn't find them.

Also we definitely need a test with two metrics of different type but with same attributes, which won't be grouped together.

Reply:
Yeah you're right: I haven't implemented tests for the non-Gauge metric types, because I wanted to make sure that everybody was on the same page on the logic before going too far in unit tests (given that the code for sums and histograms is perfectly identical).

Reply:
I will update with more unit tests.

Reply:
Tests updated! (a lot more to cover different cases, which helped me squash an undetected bug)